azavea/echo-locator
# Update neighborhood data

## Overview

This PR:
- adds the updated `neighborhood.csv`
- updates the `neighborhood_data/update_data.sh` script to remove unnecessary steps
- updates the `scripts/imagepublish` script to not upload `.gitkeep` to S3

Some things not visible in the code change:
1. `neighborhood.json` and `neighborhood_bounds.json`, which are results of the `neighborhood_data/update_data.sh` script were pushed to the data bucket on S3
2. the updated images are pushed to the staging site bucket under `assets/neighborhoods`

### Checklist


### Demo

This was the output from running the management script in my local dev database. The neighborhood records are updated:

```
$ AWS_PROFILE=echo-locator ./scripts/manage import_neighborhood_data \
>     s3://echolocator-production-data-us-east-1/neighborhoods.json \
>     s3://echolocator-production-data-us-east-1/neighborhood_bounds.json \
>     s3://echo-locator-staging-site-us-east-1/assets/neighborhoods/
Starting echo-locator_database_1 ... done
Creating echo-locator_django_run ... done
Done!

0 neighborhoods created, 236 updated
```

### Notes

1. The shape of the updated `neighborhood.csv` is consistent with the previous version. Thus, no script change needed
2. I plan to perform the following if the changes here look good and works well in the reviewer's local dev DB:
    - upload the images to the production site bucket under `assets/neighborhoods`
    - run the management script `import_neighborhood_data` on staging
    - add the corresponding steps to https://github.com/azavea/echo-locator/issues/502


## Testing Instructions

 * Pull down this branch, and run `./update_data.sh` in `neighborhood_data` directory. Make sure the neighborhood related JSON and images are generated correctly without terminal errors
 * Run `./scripts/imagepublish staging` and make sure the images are pushed to the staging site bucket under `assets/neighborhoods` prefix correctly and without the `.gitkeep` file
 * Run the following management script to update the neighborhood data in your local dev DB.
```
AWS_PROFILE=echo-locator ./scripts/manage import_neighborhood_data \
>     s3://echolocator-production-data-us-east-1/neighborhoods.json \
>     s3://echolocator-production-data-us-east-1/neighborhood_bounds.json \
>     s3://echo-locator-staging-site-us-east-1/assets/neighborhoods/
```
 * Spin up the app and browse some neighborhood info on the frontend, spot check some and make sure they still show up fine

Resolves https://github.com/azavea/echo-locator/issues/563
Resolves https://github.com/azavea/echo-locator/issues/562
